### ISSUE TITLE
DetermineBonds fails for single H atom

### ISSUE THREAD

**Issue by gncs** [rdkit/rdkit#6121]
**Describe the bug**
`rdDetermineBonds.DetermineBonds` fails for H atom and throws:
```text
ValueError: Final molecular charge does not match input; could not find valid bond ordering
```

**To Reproduce**
```python
from rdkit.Chem import rdDetermineBonds, AllChem

xyz = "1\n\nH 0.0 0.0 0.0"

mol = AllChem.MolFromXYZBlock(xyz)

rdDetermineBonds.DetermineBonds(mol, charge=0)
```

**Expected behavior**
I would expect this function to work for a single H atom. What am I missing? Thank you!

**Configuration (please complete the following information):**
 - RDKit version: 2022.9.4
 - OS: Ubuntu 20.04
 - Python version (if relevant): 3.8.15
 - Are you using conda? No
 - If you are using conda, which channel did you install the rdkit from?
 - If you are not using conda: how did you install the RDKit? pip

**Comment by dessygil** [comment]
I'm new to open-source source but I would like to give this bug a try!

**Comment by dessygil** [comment]
@gncs is it possible that the charge is supposed to be +1 since it's technically a proton? I tried this using the same code but just switching...

`rdDetermineBonds.DetermineBonds(mol, charge=1)`

let me know what your thoughts are!

**Comment by gncs** [comment]
Hi @dessygil, thank you for your interest.

Good question. In my view, the hydrogen atom (charge=0) is a valid chemical system and therefore I would expect that to work, too.

**Comment by greglandrum** [comment]
@gncs I agree: this should not happen.

**Comment by dessygil** [comment]
Okay, I would like to keep on digging into this issue.

**Comment by gncs** [comment]
Great! Let me know if anything is unclear and I'll try and see if I can help.

**Comment by dessygil** [comment]
I have it figured out I think, but I've been stuck trying to get the code running and tested for the past little while. I was wondering if there are any resources that I could reference? 

After, I think I have the workflow figured out for a PR when I'm done so I should be good. This was the code I planned to change.

`int getFormalCharge(const ROMol &mol) {
  int accum = 0;
  for (ROMol::ConstAtomIterator atomIt = mol.beginAtoms();
       atomIt != mol.endAtoms(); ++atomIt) {
    accum += (*atomIt)->getFormalCharge();
  }
  return accum;
};`


`int getFormalCharge(const ROMol &mol) {
  int accum = 0;

  if(mol->getNumAtoms() > 1){
    for (ROMol::ConstAtomIterator atomIt = mol.beginAtoms();
        atomIt != mol.endAtoms(); ++atomIt) {
      accum += (*atomIt)->getFormalCharge();
    }
  }
  return accum;
};`

**Comment by greglandrum** [comment]
Another way of thinking about this particular bug is a bit simpler: determine bonds doesn’t need to do anything for an single-atom molecule, so it can just return.

**Comment by greglandrum** [comment]
Your suggestion would return incorrect charges for single atoms ions, like H+ or Cl- or Na+

**Comment by greglandrum** [comment]
Closed in #6171
